### PR TITLE
MGMT-705 Choose boot device for the installation, taking first one

### PR DESCRIPTION
**Pull request by tsorya**
Adding boot device selection.
For now it will take the first one of type disk with lowest name, to match sda disk

### CODE REVIEW

**Comment by tsorya** [comment]
@avishayt this is done for now. We need it to go ahead with installation on baremetal.
When we will add api to choose disk, lets just change this function.

**Comment by tsorya** [comment]
as the first step the idea was to return sda in baremetal and vda for vms.
Till now we have vda hardcoded

**Comment by tsorya** [comment]
maybe something like this is better ? 
https://github.com/filanov/bm-inventory/pull/73

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -472,27 +473,33 @@ func (b *bareMetalInventory) addInstallCommand(ctx context.Context, masterNodesI
...
 		}
 		data["ROLE"] = role
-
+		disks, err := hardware.GetHostDisks(cluster.Hosts[i])
+		if err != nil {
+			return err
+		}
+		data["BOOT_DEVICE"] = fmt.Sprintf("/dev/%s", disks[0].Name)
 		buf := &bytes.Buffer{}
+		log.Debugf("Start executing install command on host %s with params %s", cluster.Hosts[i].HostID, data)
```
Comment (eranco74) [inline internal/bminventory/inventory.go:502]: Please remove this log.
We are not executing anything on the host.

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -472,27 +473,33 @@ func (b *bareMetalInventory) addInstallCommand(ctx context.Context, masterNodesI
...
 	for i := range cluster.Hosts {
 		role := cluster.Hosts[i].Role
 		if cluster.Hosts[i].ID == bootstrapId {
 			role = bootstrap
 		}
 		data["ROLE"] = role
-
+		disks, err := hardware.GetHostDisks(cluster.Hosts[i])
+		if err != nil {
+			return err
```
Comment (eranco74) [inline internal/bminventory/inventory.go:498]: Log an error for a specific host.

File: internal/hardware/hardware_utils.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		return nil, err
+	}
+	disks := getDisksList(hwInfo)
+	if len(disks) == 0 {
+		return nil, fmt.Errorf("host %s doesn't have disks", host.HostID)
+	}
+	return disks, nil
+}
+
+func getDisksList(hwInfo models.Introspection) []*models.BlockDevice {
```
Comment (eranco74) [inline internal/hardware/hardware_utils.go:23]: Just call it listDisks

**Comment by avishayt** [comment]
> maybe something like this is better ?
> #73

Yes that is awesome - thanks!

**Comment by tsorya** [comment]
@avishayt i will reopen it with changes

**Pull request closed without merging**